core/src/main/java/nu/yona/server/analysis/service/AnalysisEngineService.java

```diff
@@ -107,7 +110,7 @@ private void updateLastGoalConflictMessage(PotentialConflictDTO payload, Date me
 		message.setEndTime(messageEndTime);
 	}
 
-	private Set<Goal> determineConflictingGoalsForUser(UserAnonymized userAnonymized, Set<String> categories)
+	private Set<Goal> determineConflictingGoalsForUser(UserAnonymizedDTO userAnonymized, Set<String> categories)
 	{
 		Set<Goal> allGoals = goalService.getAllGoalEntities();
```

Comment:
As a next step, we should change this too. The goal service should return goal DTOs and cache them.